codice/ddf
# [2.13.x] DDF-4424 Updating the CDM File Directory (in-place) to update the observer state only on a successful ingest to the catalog framework

#### What does this PR do?
Fixes a problem where the CDM would update it's internal state regardless of if the process completed
successfully or not. This could lead to the CDM and catalog getting out of sync when a network disconnect or ingest failure occurred.

#### Who is reviewing it? 

@peterhuffer 
@kcover 

#### Select relevant component teams: 

#### Ask 2 committers to review/merge the PR and tag them here.

@lessarderic 
@clockard 
@vinamartin 

#### How should this be tested?
1. Install DDF
2. Install the standard profile and turn logs up on the CDM. 
3. `log:set TRACE org.codice.ddf.catalog.content.monitor`
** I found it easier to test the disconnect with a breakpoint in the code but It's not required with a big enough directory.
4. start the CDM (in-place) monitoring a network file system. /admin -> catalog -> configurations -> cdm
5. disconnect from the network file system and check that things failed ingesting.
6. Reconnect to the network file system.
7. ensure that everything resumed and finished ingesting.


#### Any background context you want to provide?
#### What are the relevant tickets?
[DDF-4424](https://codice.atlassian.net/browse/DDF-4424)
#### Screenshots

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
